rest_framework/compat.py

```diff
@@ -32,7 +32,7 @@
 def get_regex_pattern(urlpattern):
     if hasattr(urlpattern, 'pattern'):
         # Django 2.0
-        return urlpattern.pattern.regex.pattern
+        return str(urlpattern.pattern)
```

Comment:
I think this has broken #5672 even further - instead of getting the expected regex it now gets a path/regex hybrid like `'convtest/<int:pk>\\.(?P<format>[a-z0-9]+)/?$'` when trying to add suffix patterns.

Reply:
Indeed, `str(urlpattern.pattern)` is not a regex pattern anymore. Maybe we need to undo this change and have an if-else in `EndpointEnumerator`, depending if urlpattern is a RoutePattern or a RegexPattern?

Reply:
That would still leave #5672 broken, just in a different way.

Perhaps renaming the method to something like `get_original_route` would help avoid confusion, but a separate fix would still be needed for the other issue.

Reply:
No problem adding a similar, not quite duplicate, function to compat. (We can always abstract again later.)